File: LAB1.py

#question 1
name = input("What is your name? ")
print("Hello " + name + ".")
x= input("What is your Student ID? ")
print("Your ID is " + x )

#question 2

seconds=int(input("ENTER TOTAL SECONDS"))
hours= seconds//3600
mins= (seconds%3600)//60
remainseconds = seconds%60
print(seconds+ "is" + hours+ "," +mins + "and" +remainseconds)


#question 3
x1= float(input("ENTER x1 :"))
y1= float(input("ENTER y1:"))
x2= float(input("ENTER x2 :"))
y2= float(input("ENTER y2 :"))

a= x2- x1
b= y2-y1

distance= (a*a + b*b) ** 0.5


#question 4
print("*******/n*****/n***/n* ")

File: lab1.cpp
#include <iostream>
#include <string>

using namespace std;
int main() {
    // question1
    string name;
    string student_id;

    cout << "What is your name?" << endl;
    cin >> name;
    cout << "Hello " << name << "." << endl;

    cout << "What is your Student ID?" << endl;
    cin >> student_id;
    cout << "Your ID is " << student_id << "." << endl;

    cout << "-----" << endl;

    // question2
    int total;
    cout << "Enter total seconds: ";
    cin >> total;

    int hours = total / 3600;
    int remaining = total % 3600;
    int minutes = remaining / 60;
    int seconds = remaining % 60;

    cout << total << " seconds is " << hours << " hours " << minutes << " minutes and " << seconds << " seconds" << endl;

    cout << "---" << endl;

    // question3
    double x1, y1, x2, y2;
    cout << "Enter x1: "; cin >> x1;
    cout << "Enter y1: "; cin >> y1;
    cout << "Enter x2: "; cin >> x2;
    cout << "Enter y2: "; cin >> y2;

    double dx = x2 - x1;
    double dy = y2 - y1;
    double square = (dx * dx) + (dy * dy);
    double a = square / 2.0;
    for(int i = 0; i < 10; i++) {
        a = (a + square/ a) / 2.0;
    }

    cout << "Distance is approximately: " << a<< endl;

    cout << "------" << endl;

    // question4

    string pattern = "";
    int star = 7;
    int gap = 0;

    while (star >= 1) {
        // Boşlukları ekle
        for (int i = 0; i <gap; i++) {
            pattern = pattern + " ";
        }
        // Yıldızları ekle
        for (int j = 0; j < star; j++) {
            pattern = pattern + "*";
        }
        // Satır sonu ekle
        pattern= pattern + "\n";

        star = star - 2; gap= gap+ 1;
    }

    cout <<pattern;

    return 0;
}